javascript/display-data.js

```diff
@@ -28,16 +27,28 @@ var populateBlogFilter = function(elem){
   }
 }
 
-Data.populatePage = function(){
-  $.get('data/data.json',function(data){
-    data.forEach(function(elem){
-      blogCollection.push(elem);
+Data.populateBlogPage = function(){
+  $.get('data/data.json',function(data,text,xhr){
+    var value = [];
+    responseETag = xhr.getResponseHeader('Etag');
+    if(localStorage.getItem('eTag')===null||responseETag!=localStorage.getItem('eTag')){
+      localStorage.setItem('eTag',responseETag);
+      localStorage.setItem('blogArticles',JSON.stringify(data));
+      value = data;
+    }
+    else{
+      value = JSON.parse(localStorage.getItem('blogArticles'));
+    }
+    var blogCollection = [];
+    value.map(function(elem){
```

Comment:
This is basically a forEach, re-written as a .map function

with the .map function you want to be returning the new array to a variable or passing it straight into another function.